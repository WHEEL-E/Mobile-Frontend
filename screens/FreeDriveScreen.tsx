import React from "react";
import {
  Dimensions,
  ImageBackground,
  StyleSheet,
  Text,
  View,
} from "react-native";
import { useTranslation } from "react-i18next";
import colors from "../constants/colors";
import DriveWheel from "../components/UI/freeDriveComponents/DriveWheel";
import fonts from "../constants/fonts";
import { BackButton } from "../components/UI/BackButton";

const FreeDriveScreen = (props: any) => {
  const { t } = useTranslation();
  return (
    <View style={styles.container}>
      <View style={styles.backButton}>
        <BackButton onPress={() => props.navigation.goBack()} />
      </View>
      <ImageBackground
        testID="backgroundImage"
        source={require("../assets/Vector.png")}
        style={styles.backgroundImage}
      >
        <Text style={styles.title}>{t("freeDrive")}</Text>
        <DriveWheel />
        <Text testID="instructionText" style={styles.text}>
          {t("freeDriveText")}
        </Text>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
  },
  backButton: {
    position: "absolute",
    top: "7%",
    left: "8%",
  },
  backgroundImage: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    resizeMode: "cover",
    padding: "10%",
  },
  title: {
    fontSize: 35,
    fontFamily: "Cairo-Bold",
  },
  text: {
    color: colors.darkGreen,
    fontFamily: fonts.CairoRegular,
    lineHeight: 20,
  },
});
export default FreeDriveScreen;
