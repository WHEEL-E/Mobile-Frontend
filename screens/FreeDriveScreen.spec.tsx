import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import * as cmp from "react-native";
import { ImageBackground } from "react-native";
import FreeDriveScreen from "./FreeDriveScreen";

// jest.mock("react-i18next", () => {
//   return {
//     useTranslation: () => {
//       return { t: jest.fn };
//     },
//   };
// });

jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: any) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe("patienthomeScreen", () => {
  it("makes sure screen matches snapShot", () => {
    const dummyProp: any = () => {};
    const freeDriveScreen = renderer
      .create(<FreeDriveScreen navigation={dummyProp} route={dummyProp} />)
      .toJSON();
    expect(freeDriveScreen).toMatchSnapshot();
  });

  it("makes sure image background is called", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <FreeDriveScreen navigation={dummyProp} route={dummyProp} />
    );
    const backgroundImage = getByTestId("backgroundImage");
    expect(backgroundImage.props.source).toBe(require("../assets/Vector.png"));
  });

  it("makes sure text is rendered properly", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <FreeDriveScreen navigation={dummyProp} route={dummyProp} />
    );
    const instructionText = getByTestId("instructionText");
    expect(instructionText.children).toContain("freeDriveText");
  });
});
