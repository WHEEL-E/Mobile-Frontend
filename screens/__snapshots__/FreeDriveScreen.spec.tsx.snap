// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patienthomeScreen makes sure screen matches snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": "10%",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="backgroundImage"
    />
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Bold",
          "fontSize": 35,
        }
      }
    >
      freeDrive
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7D9EA",
          "borderRadius": 337.5,
          "height": 675,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 50,
          "width": 675,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4278190080,
              4282992969,
              4278190080,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 300,
            "height": 600,
            "justifyContent": "center",
            "width": 600,
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4278190080,
                  4278190080,
                  4287927444,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 225,
                "height": 450,
                "justifyContent": "center",
                "width": 450,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4287927444,
                    4278190080,
                    4278190080,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": undefined,
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 150,
                  "height": 300,
                  "justifyContent": "center",
                  "width": 300,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                proxiedProperties={
                  Object {
                    "colors": Array [
                      4287927444,
                      4278190080,
                      4278190080,
                      4287927444,
                    ],
                    "endPoint": undefined,
                    "locations": undefined,
                    "startPoint": undefined,
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 75,
                    "height": 150,
                    "justifyContent": "center",
                    "width": 150,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 9.375,
                      "height": 18.75,
                      "width": 18.75,
                    }
                  }
                />
              </ViewManagerAdapter_ExpoLinearGradient>
            </ViewManagerAdapter_ExpoLinearGradient>
          </ViewManagerAdapter_ExpoLinearGradient>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <Text
      style={
        Object {
          "color": "#11698E",
          "fontFamily": "Cairo-Regular",
          "lineHeight": 20,
        }
      }
      testID="instructionText"
    >
      freeDriveText
    </Text>
  </View>
</View>
`;
